client/src/main/java/org/apache/uniffle/client/request/CreateShuffleReadClientRequest.java

```diff
@@ -45,6 +45,44 @@ public class CreateShuffleReadClientRequest {
   private IdHelper idHelper;
   private ShuffleDataDistributionType shuffleDataDistributionType = ShuffleDataDistributionType.NORMAL;
   private boolean expectedTaskIdsBitmapFilterEnable = false;
+  private int prefetchBatchNum;
+
+  public CreateShuffleReadClientRequest(
```

Comment:
> Async read will cost more memory?

Yes, the original occupied memory size depends on the `readBufferSize.` And the size of this PR is `prefetchBatchNum*readBufferSize`

> Do we need use Spark MemoryManager to manage our read buffer?

This part is the general client for MR/Spark/TeZ, which is not bound to Spark.

Reply:
For Spark, we can manage it in RssShuffleReader.

Reply:
Maybe we could introduce a uniffle client side memoryManager. In spark, this could be extended with Spark's MemoryManager and then the memory could be required by this before fetching more data in async data loader.

And for MR or other frameworks, the memoryManager could be bypass. But it seems this is a huge feature and need to discuss more.

Reply:
It's ok that only Spark support Memory Manager.

Reply:
I think we should add a switch for this feature. Because costing more memory will cause existing jobs to fail.

Reply:
Yes, this is a potential risk point, but the memory consumed is decided to the `prefetchBatchNum*readBufferSize`, and by default, and it's acceptable by using default value from conf.